smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/HttpBindingProtocolGenerator.java

```diff
@@ -284,70 +266,133 @@ private void generateOperationSerializer(
         return queryBindings;
     }
 
+    private void writeHeaders(
```

Comment:
This method was moved from below for consistency, and also has fixes to properly write header and prefix header contents.